x-pack/plugins/fleet/server/routes/package_policy/handlers.ts

```diff
@@ -398,22 +400,49 @@ export const deletePackagePolicyHandler: RequestHandler<
   const soClient = coreContext.savedObjects.client;
   const esClient = coreContext.elasticsearch.client.asInternalUser;
   const user = appContextService.getSecurity()?.authc.getCurrentUser(request) || undefined;
+  const logger = appContextService.getLogger();
+
   try {
-    const body: DeletePackagePoliciesResponse = await packagePolicyService.delete(
+    try {
+      const packagePolicies = await Promise.all(
+        request.body.packagePolicyIds.map(async (id) => {
+          try {
+            return await packagePolicyService.get(soClient, id);
```

Comment:
We have a `packagePolicyService.getByIDs` method that accept a ignoreMissing params that will do a bulk get that could be more interesting to use here

Reply:
Done.